WcaOnRails/app/models/user.rb

```diff
@@ -428,7 +428,7 @@ def can_view_hidden_competitions?
   end
 
   def can_edit_registration?(registration)
-    can_manage_competition?(registration.competition) || (!registration.accepted? && registration.user_id == self.id)
+    can_manage_competition?(registration.competition) || (!registration.accepted? && !registration.competition.registration_past? && registration.user_id == self.id)
```

Comment:
This line makes me sad. So long. And I don't understand it.

Maybe there isn't much to do about it. That's OK in that case.